File: app/src/main/java/com/nurram/project/imagetextrecognition/room/WordRoomDatabase.java
package com.nurram.project.imagetextrecognition.room;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import android.content.Context;

@Database(entities = {Word.class}, version = 1)
abstract class WordRoomDatabase extends RoomDatabase {
    abstract WordDao mWordDao();

    private static volatile WordRoomDatabase sDatabase;

    static WordRoomDatabase getDatabase(Context context) {
        if (sDatabase == null) {
            synchronized (WordRoomDatabase.class) {
                if (sDatabase == null) {
                    sDatabase = Room.databaseBuilder(context.getApplicationContext(),
                            WordRoomDatabase.class, "word_database")
                            .build();
                }
            }
        }

        return sDatabase;
    }
}

File: app/src/main/java/com/nurram/project/imagetextrecognition/room/WordRoomDatabase.kt
package com.nurram.project.imagetextrecognition.room

import android.app.Application
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase

@Database(entities = [Word::class], version = 1)
abstract class WordRoomDatabase : RoomDatabase() {
    abstract fun mWordDao(): WordDao

    companion object {
        @Volatile
        private var db: WordRoomDatabase? = null

        fun getDatabase(application: Application): WordRoomDatabase? {
            if (db == null) {
                synchronized(WordRoomDatabase::class.java) {
                    if (db == null) {
                        db = Room.databaseBuilder(
                            application,
                            WordRoomDatabase::class.java, "pronunhelper_db"
                        )
                            .fallbackToDestructiveMigration()
                            .build()
                    }
                }
            }

            return db
        }
    }
}